packages/medusa/src/migrations/1649775522087-add_batch_job_model.ts

```diff
@@ -5,15 +5,42 @@ export class addBatchJobModel1649775522087 implements MigrationInterface {
 
   public async up(queryRunner: QueryRunner): Promise<void> {
     await queryRunner.query(
-      `CREATE TYPE "batch_job_status_enum" AS ENUM('created', 'processing', 'awaiting_confirmation', 'completed')`
+      `CREATE TYPE "batch_job_status_enum" AS ENUM('created', 'processing', 'awaiting_confirmation', 'canceled', 'completed')`
     )
+
+    await queryRunner.query(
+        `CREATE TABLE "batch_job"
+         (
+             "id"                       character varying                NOT NULL,
+             "type"                     text                             NOT NULL,
+             "status"                   "public"."batch_job_status_enum" NOT NULL,
```

Comment:
we should remove the status field from the batch job db entity, and infer the status by the timestamp set